Sharding  not support ZoneDateTime

## Sharding  not support ZoneDateTime


### Which version of ShardingSphere did you use?

5.5.1

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

ShardingSphere-JDBC 

### Expected behavior

zoneDateTime can be parsed correctly



### Actual behavior

The field was parsed as LocalDateTime, which subsequently caused a conversion exception.

```
Caused by: java.lang.ClassCastException: class java.time.LocalDateTime cannot be cast to class java.time.ZonedDateTime (java.time.LocalDateTime and java.time.ZonedDateTime are in module java.base of loader 'bootstrap')
	at org.apache.ibatis.type.ZonedDateTimeTypeHandler.getNullableResult(ZonedDateTimeTypeHandler.java:39)
	at org.apache.ibatis.type.ZonedDateTimeTypeHandler.getNullableResult(ZonedDateTimeTypeHandler.java:29)
	at org.apache.ibatis.type.BaseTypeHandler.getResult(BaseTypeHandler.java:86)
	... 157 more
```

### Reason analyze (If you can)
org.apache.shardingsphere.infra.executor.sql.execute.result.query.impl.driver.jdbc.type.stream.JDBCStreamQueryResult#getValue

<img width="875" alt="image" src="https://github.com/user-attachments/assets/047b5a4a-7a2c-4f1f-abec-6687029a6b42">
should be 

```java
 return resultSet.getObject(columnIndex, type);
```

<img width="1600" alt="image" src="https://github.com/user-attachments/assets/30749f30-af14-4e82-a7df-3d9f804fb43d">

The type-specified resultSet.getObject supports parsing ZonedDateTime. However, in  `org.apache.shardingsphere.infra.executor.sql.execute.result.query.impl.driver.jdbc.type.stream.JDBCStreamQueryResult#getValue`, the method’s last line calls return resultSet.getObject(columnIndex); without specifying a type, resulting in ZonedDateTime parsing not being applied.

As a result, this method defaults to parsing database fields with dateTime types as LocalDateTime.

--- a/infra/executor/src/main/java/org/apache/shardingsphere/infra/executor/sql/execute/result/query/impl/driver/jdbc/type/stream/JDBCStreamQueryResult.java
+++ b/infra/executor/src/main/java/org/apache/shardingsphere/infra/executor/sql/execute/result/query/impl/driver/jdbc/type/stream/JDBCStreamQueryResult.java
@@ -104,7 +104,7 @@ public final class JDBCStreamQueryResult extends AbstractStreamQueryResult {
         if (Array.class == type) {
             return resultSet.getArray(columnIndex);
         }
-        return resultSet.getObject(columnIndex);
+        return resultSet.getObject(columnIndex, type);
     }
     
     @Override

--- a/infra/executor/src/main/java/org/apache/shardingsphere/infra/executor/sql/execute/result/query/impl/driver/jdbc/type/stream/JDBCStreamQueryResult.java
+++ b/infra/executor/src/main/java/org/apache/shardingsphere/infra/executor/sql/execute/result/query/impl/driver/jdbc/type/stream/JDBCStreamQueryResult.java
@@ -34,6 +34,7 @@ import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Time;
 import java.sql.Timestamp;
+import java.time.ZonedDateTime;
 import java.util.Calendar;
 
 /**
@@ -104,7 +105,10 @@ public final class JDBCStreamQueryResult extends AbstractStreamQueryResult {
         if (Array.class == type) {
             return resultSet.getArray(columnIndex);
         }
-        return resultSet.getObject(columnIndex, type);
+        if (ZonedDateTime.class == type) {
+            return resultSet.getObject(columnIndex, type);
+        }
+        return resultSet.getObject(columnIndex);
     }
     
     @Override

--- a/infra/executor/src/test/java/org/apache/shardingsphere/infra/executor/sql/execute/result/query/impl/driver/jdbc/type/stream/JDBCStreamQueryResultTest.java
+++ b/infra/executor/src/test/java/org/apache/shardingsphere/infra/executor/sql/execute/result/query/impl/driver/jdbc/type/stream/JDBCStreamQueryResultTest.java
@@ -28,6 +28,9 @@ import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Time;
 import java.sql.Timestamp;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
 import java.util.Calendar;
 
 import static org.hamcrest.CoreMatchers.is;
@@ -163,6 +166,13 @@ class JDBCStreamQueryResultTest {
         ResultSet resultSet = mock(ResultSet.class);
         when(resultSet.getTimestamp(1)).thenReturn(new Timestamp(0L));
         assertThat(new JDBCStreamQueryResult(resultSet).getValue(1, Timestamp.class), is(new Timestamp(0L)));
+    }
+
+    @Test
+    void assertGetValueByZonedDateTime() throws SQLException {
+        ResultSet resultSet = mock(ResultSet.class);
+        when(resultSet.getObject(1, ZonedDateTime.class)).thenReturn(ZonedDateTime.ofInstant(Instant.ofEpochSecond(1), ZoneId.systemDefault()));
+        assertThat(new JDBCStreamQueryResult(resultSet).getValue(1, ZonedDateTime.class), is(ZonedDateTime.ofInstant(Instant.ofEpochSecond(1), ZoneId.systemDefault())));
     }
     
     @Test

--- a/infra/executor/src/test/java/org/apache/shardingsphere/infra/executor/sql/execute/result/query/impl/driver/jdbc/type/stream/JDBCStreamQueryResultTest.java
+++ b/infra/executor/src/test/java/org/apache/shardingsphere/infra/executor/sql/execute/result/query/impl/driver/jdbc/type/stream/JDBCStreamQueryResultTest.java
@@ -167,7 +167,7 @@ class JDBCStreamQueryResultTest {
         when(resultSet.getTimestamp(1)).thenReturn(new Timestamp(0L));
         assertThat(new JDBCStreamQueryResult(resultSet).getValue(1, Timestamp.class), is(new Timestamp(0L)));
     }
-
+    
     @Test
     void assertGetValueByZonedDateTime() throws SQLException {
         ResultSet resultSet = mock(ResultSet.class);

